packages/account/src/BiconomySmartAccount.ts

```diff
@@ -288,10 +288,14 @@ export class BiconomySmartAccount extends SmartAccount implements IBiconomySmart
     } catch (error) {
       // Not throwing this error as nonce would be 0 if this.nonce() throw exception, which is expected flow for undeployed account
     }
+    let isDeployed = true
+
+    isDeployed = nonce.eq(0) ? await this.isAccountDeployed(this.address) : true
```

Comment:
```
if(nonce.eq(0)) {
 isDeployed = await this.isAccountDeployed(this.address)
}
```

above code would be more readable. lmk what you think

Reply:
> ```
> if(nonce.eq(0)) {
>  isDeployed = await this.isAccountDeployed(this.address)
> }
> ```
> 
> above code would be more readable. lmk what you think

yeah, i think yours approach is more readable. I was thinking of multiple cases and used ternary but removed other cases as they were not valid and did not changed the logic from ternary approach to if statement